core/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -241,6 +246,58 @@ static MetaData updateMetaDataWithRoutingTable(MetaData oldMetaData, RoutingTabl
     }
 
     /**
+     * Possibly removes delay markers from shards and reroutes if markers have been deleted.
+     * <p>
+     * If the same instance of the routing table is returned, then no change has been made.</p>
+     */
+    public RoutingAllocation.Result applyDelayedShards(ClusterState clusterState) {
+        return applyDelayedShards(clusterState, true);
+    }
+
+    public RoutingAllocation.Result applyDelayedShards(ClusterState clusterState, boolean withReroute) {
+        RoutingNodes routingNodes = getMutableRoutingNodes(clusterState);
+        RoutingAllocation allocation = new RoutingAllocation(allocationDeciders, routingNodes, clusterState.nodes(), clusterInfoService.getClusterInfo(), currentNanoTime());
+        boolean changed = removeDelayMarkers(allocation);
```

Comment:
it's different because applyStartedShards and  applyFailedShard are doing something that isn't done in the reroute. Also this method is called when we are 99% sure that something is going to happen.  I don't think it's worth the extra code path.